src/main/java/javaserver/ResponseBuilderFactory.java

```diff
@@ -0,0 +1,58 @@
+package javaserver;
```

Comment:
The name of this class can be a little confusing. By glancing at this class without diving into the implementation, I can't tell if it's something that implements the Builder creational design pattern, Factory pattern, or if its an interesting combination of both.